codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/ApiGatewayExportsNullabilityExceptionIntegration.java

```diff
@@ -0,0 +1,191 @@
+            ShapeId.from("com.amazonaws.kafka#Kafka"),
+            ShapeId.from("com.amazonaws.mediapackagevod#MediaPackageVod"),
+            ShapeId.from("com.amazonaws.mq#mq"),
+            ShapeId.from("com.amazonaws.iot1clickdevicesservice#IoT1ClickDevicesService"),
+            ShapeId.from("com.amazonaws.serverlessapplicationrepository#ServerlessApplicationRepository"),
+            ShapeId.from("com.amazonaws.schemas#schemas"),
+            ShapeId.from("com.amazonaws.pinpointsmsvoice#PinpointSMSVoice"),
+            ShapeId.from("com.amazonaws.route53recoveryreadiness#Route53RecoveryReadiness"),
+            // APIGW services migrated to Smithy
+            ShapeId.from("com.amazonaws.dataexchange#DataExchange"),
+            ShapeId.from("com.amazonaws.kafkaconnect#KafkaConnect"),
+            ShapeId.from("com.amazonaws.mediatailor#MediaTailor"));
+
+    @Override
+    public Model preprocessModel(Model model, GoSettings settings) {
+        ShapeId service = settings.getService();
+        if (!APIGW_NULLABILITY_SERVICES.contains(service)) {
+            return model;
+        }
+        try {
+            return handleApiGateWayExportsNullabilityExceptions(model, service);
+        } catch (Exception e) {
+            LOGGER.severe("Failed to handle APIGW exports nullability exceptions for service: " + service.toString());
+            System.err.println(e);
+            e.printStackTrace();
+            return model;
+        }
+    }
+
+    private Model handleApiGateWayExportsNullabilityExceptions(Model model, ShapeId service) throws IOException {
+        LOGGER.info("Handling APIGW exports nullability exceptions for service: " + service.toString());
+
+        // Read ShapeId exceptions
+        List<ShapeId> shapeIdsToReplace = new ArrayList<>();
+        ObjectNode node = Node
+                .parse(Files.readString(Path.of(PATH_PREFIX, service.toString() + ".exceptions.json")))
+                .expectObjectNode();
+        for (StringNode shapeIdNode : node.expectArrayMember("picked").getElementsAs(StringNode.class)) {
+            shapeIdsToReplace.add(ShapeId.from(shapeIdNode.getValue()));
+        }
+
+        // Get Shapes to replace
+        List<Shape> shapesToReplace = new ArrayList<>();
+        Set<BooleanShape> booleanShapes = model.getBooleanShapes();
+        Set<NumberShape> numberShapes = model.toSet(NumberShape.class);
+        Set<MemberShape> memberShapes = model.getMemberShapes();
+
+        for (ShapeId shapeId : shapeIdsToReplace) {
+            Optional<Shape> optionalShape = model.getShape(shapeId);
+            if (!optionalShape.isPresent()) {
+                LOGGER.severe("APIGW exports nullability exception could not be found in model: " + shapeId.toString());
+                continue;
+            }
+            Shape shape = optionalShape.get();
+            // Boolean shape
+            if (booleanShapes.contains(shape)) {
```

Comment:
Note that when you add a default to root-level shapes, you have to find all members that target it so they also get the default. You can't only rely on a snapshot of members in the JSON file for this, since new members could be added that target one of these root level shapes that you're adding a default to. The members need to repeat the default to be valid.

It might be useful to also validate the model after patching these defaults back in.